### PR TITLE
Added CheckNameAvailability, Identity, systemData, used ErrorResponse v2 in DeviceUpdate

### PR DESCRIPTION
**Pull request by abatishchev**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [x] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by Sedols — APPROVED** [review]

**Comment by abatishchev** [comment]
@raych1 I see two failure I cannot explain:

- SDK azure-sdk-for-net failed with
```
/home/vsts/work/1/s/azure-sdk-for-net/build.proj : error MSB4057: The target "CreateNugetPackage" does not exist in the project.
```

- Swagger PrettierCheck failed with
```
prettier -c specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/examples/CheckNameAvailability_Available.json
[warn] specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/examples/CheckNameAvailability_AlreadyExists.json
[warn] Code style issues found in the above file(s). Forgot to run Prettier?

Checking formatting...

prettier -c specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/examples/Instances/Instances_Create.json
[warn] specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/examples/CheckNameAvailability_Available.json
[warn] Code style issues found in the above file(s). Forgot to run Prettier?
```

**Review by Sedols — APPROVED** [review]

File: specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json [hunk 1]
```diff
@@ -271,7 +317,13 @@
         ],
         "responses": {
           "200": {
-            "description": "Account updated successfully.",
+            "description": "Account was updated successfully.",
+            "schema": {
+              "$ref": "#/definitions/Account"
+            }
+          },
+          "201": {
```
Comment (varunkumta) [inline specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json:325]: >201 [](start = 11, length = 3)

a patch should not return a 201, but a 202 to indicate async completion

File: specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json [hunk 2]
```diff
@@ -682,6 +746,33 @@
         }
       }
     },
+    "Identity": {
```
Comment (varunkumta) [inline specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json:749]: this is available in common-types, any reason to not re-use that?

File: specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/examples/CheckNameAvailability_Available.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    "request": {
+      "name": "contoso",
+      "type": "Microsoft.DeviceUpdate/accounts"
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "nameAvailable": true,
+        "reason": "NotSpecified"
```
Comment (varunkumta) [inline specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/examples/CheckNameAvailability_Available.json:15]: >NotSpecified [](start = 19, length = 12)

remove

File: specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json (hunk 1 above)
Comment (abatishchev) [reply specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json:325]: Usually PUT returns 201 and DELETE returns 202. In our case PATCH initiates the same flow as PUT hence 201.

File: specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/examples/CheckNameAvailability_AlreadyExists.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+  "parameters": {
+    "api-version": "2020-03-01-preview",
+    "subscriptionId": "00000000-0000-0000-0000-000000000000",
+    "locationName": "westus2",
```
Comment (varunkumta) [inline specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/examples/CheckNameAvailability_AlreadyExists.json:5]: this property is not in the swagger - remove

File: specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/examples/Accounts/Accounts_Update.json [hunk 5]
```diff
@@ -25,6 +25,23 @@
           "tagKey": "tagValue"
         }
       }
+    },
+    "201": {
```
Comment (varunkumta) [inline specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/examples/Accounts/Accounts_Update.json:29]: >1 [](start = 7, length = 1)

202 - no response body

File: specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json (hunk 1 above)
Comment (varunkumta) [reply specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json:325]: 201 is Created, a PATCH cannot be used to create a resource - and hence it is not a valid status code to return

File: specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/examples/CheckNameAvailability_AlreadyExists.json (hunk 4 above)
Comment (abatishchev) [reply specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/examples/CheckNameAvailability_AlreadyExists.json]: Removed from both

File: specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json (hunk 2 above)
Comment (Sedols) [reply specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json:749]: The common type includes identity type enums we do not support at the moment like "UserAssigned" and "SystemAssigned, UserAssigned"

Comment (Sedols) [reply specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json:749]: Also I think the common definition did no include "None" which is a bug.

File: specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json (hunk 1 above)
Comment (Sedols) [reply specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json:325]: We are behind RPaaS and currently they do not support 202 for async operation only 201 even for patch. We reached out to them and they said they will work either to introduce 202 or give guidance to accept 201 for there reviews in the future. For now here is the documentation https://armwiki.azurewebsites.net/rpaas/async.html If there are any questions please reach out to RPaaS Team.

File: specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json (hunk 2 above)
Comment (abatishchev) [reply specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json:749]: Both common types v1 and v2 contain single enum value: SystemAssigned
See https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json#L238

I can submit a PR to extend that. What should be the values?

```
None
SystemAssigned
UserAssigned
SystemAssigned, UserAssigned
```

Or how Swagger support multiple values?

File: specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/examples/CheckNameAvailability_Available.json (hunk 3 above)
Comment (abatishchev) [reply specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/examples/CheckNameAvailability_Available.json:12]: I tried to call ARM for my type, which then I'd handled by RPaaS. That was the response.

Is it incorrect? Should I work with RPaaS to remove `reason` on success?

File: specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json (hunk 2 above)
Comment (varunkumta) [reply specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json:749]: please leave as-is for now, we are discussing making the update to the common-types files to support UserAssigned and None

File: specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json (hunk 1 above)
Comment (varunkumta) [reply specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2020-03-01-preview/deviceupdate.json:325]: I confirmed with the RPaaS team that this is an issue today, so I will sign off. However they are looking at changing it so please work with them as to what they want you to do for this PR

**Review by raych1 — APPROVED** [review]

**Comment by raych1** [comment]
Breaking change review policy is out of scope since this service isn't GAed.

**Pull request merged**